great_expectations/data_context/store/gx_cloud_store_backend.py

```diff
@@ -543,20 +543,48 @@ def remove_key(self, key):
             )
 
     def _update(self, key, value, **kwargs):
-        existing = self._get(key)
+        existing = self._get(key)["data"]
+
+        # if the provided key does not contain id (only name), cloud will return a list of resources filtered
+        # by name, with length >= 0, instead of a single object (or error if not found)
+        if isinstance(existing, list) and len(existing) > 1:
```

Comment:
Hmm is there any way we can do this without specifically `isinstance` checking? I think this logic makes sense but just trying to brainstorm

Reply:
also do our `TypedDict`'s need to be updated at all for this singular vs multiple resource logic? Can we be consistent and always return a list or does that not make sense?

Reply:
Which TypedDicts are you referring to? `ResponsePayload`? Hm, looking at `ResponsePayload`, looks like the `data` field is incorrect. It currently says `data` field is `PayloadDataField`, but it can also be `list[PayloadDataField]` - does that annotation make sense?

Reply:
Don't think it makes sense to always return list, since list only makes sense if cloud is returning a collection.